win_etw_macros/src/lib.rs

```diff
@@ -409,6 +414,41 @@ fn trace_logging_events_core(attr: TokenStream, item_tokens: TokenStream) -> Tok
         let event_level = event_attrs.level;
         let event_opcode = event_attrs.opcode;
         let event_task = event_attrs.task;
+        let potential_event_id = event_attrs.event_id;
+
+        // We use the first entry to see if we have user provided IDs
+        // or we are generating one.
+        if event_index == 0 {
+            event_ids_auto_generated = potential_event_id.is_none();
+        }
+
+        // We have some events with user provided id and some without.
+        if event_ids_auto_generated != potential_event_id.is_none() {
+            errors.push(Error::new(
+                method.span(),
+                "Event ids must be set for all events, or for none.",
+            ));
+        }
+
+        let event_id = potential_event_id.unwrap_or(event_index);
```

Comment:
Is this the only place that `event_id` is used? If so, then this `unwrap_or` and the `if` statement below it should be combined, like so:

```
if let Some(event_id) = potential_event_id {
   ... use event_id ...
} else {
  ... nope, no event id was specified ...
}
```